libraries/standard/mqtt/src/mqtt_state.c

```diff
@@ -182,13 +295,45 @@ static bool validateTransitionAck( MQTTPublishState_t currentState,
             break;
 
         case MQTTPubRelPending:
-            /* Incoming publish, QoS 2. */
-            isValid = ( newState == MQTTPubCompSend ) ? true : false;
+
+            /* Incoming publish, QoS 2.
+             * There are 2 valid transitions possible.
+             * 1. MQTTPubRelPending -> MQTTPubCompSend : A PUBREL ack is received
+             *    when publish record state is MQTTPubRelPending. This is the
+             *    normal state transition without any connection interuptions.
+             * 2. MQTTPubRelPending -> MQTTPubRelPending : Receiving a duplicate
+             *    QoS2 publish can result in this no transition.
```

Comment:
"this no transition" -> "a transition to the same state"

Reply:
Thanks